### PR TITLE
Implement all methods declared in the C++ headers

### PR DESCRIPTION
**Pull request by saraedum**
I had forgotten about a few in the first run. The only non-trivial
new implementation is the operator bool. Everything else just moved from
the now deprecated implementation to the non-deprecated one.

### CODE REVIEW

**Comment by saraedum** [comment]
I reordered the implementations the same way they are ordered in the header file.

File: renfxx/renf_elem_class.cpp [hunk 1]
```diff
@@ -322,44 +239,92 @@ mpz_class renf_elem_class::get_num(void) const
...
+    mpz_class x;
+    std::vector<mpz_class> res;
+
+    if (nf == NULL)
+    {
+        fmpz_get_mpz(x.__get_mp(), fmpq_numref(b));
+        res.push_back(x);
+    }
     else
     {
```
Comment (saraedum) [inline renfxx/renf_elem_class.cpp]: oops

File: e-antic/renfxx.h [hunk 2]
```diff
@@ -35,13 +35,11 @@ class renf_class : boost::equality_comparable<renf_class> {
     renf_class() noexcept;
     renf_class(const renf_class &) noexcept;
     renf_class(const ::renf_t &, const std::string & gen = "a") noexcept;
-    renf_class(::renf_t &&, const std::string & gen = "a") noexcept;
```
Comment (saraedum) [inline e-antic/renfxx.h]: I dropped the move operators. There's no `renf_swap` (I guess) and it does not seem to be worth it.

File: e-antic/renfxx.h [hunk 3]
```diff
@@ -144,9 +142,9 @@ class renf_elem_class : boost::ordered_field_operators<renf_elem_class>,
     ::renf_elem_t & renf_elem_t() const noexcept;
 
     // data conversion
-    mpz_class den() const;
-    mpz_class num() const;
-    explicit operator mpq_class() const;
```
Comment (saraedum) [inline e-antic/renfxx.h]: It's a fatal exception to call these when the entities do not make sense. Not an exception that could be could ⇒ `noexcept`.

File: e-antic/renfxx.h (hunk 2 above)
Comment (videlec) [reply e-antic/renfxx.h]: indeed

File: e-antic/renfxx.h (hunk 3 above)
Comment (videlec) [reply e-antic/renfxx.h]: "could be could"?

**Pull request merged**